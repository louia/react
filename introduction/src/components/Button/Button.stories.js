import React from 'react';
import Button from './index';
import '../../index.css'

export default {
    title: 'components/Button',
    component: Button,
}

export const default_value = () => <Button />;
export const Button_with_text_children = () => <Button>Mon Bouton</Button>;
export const Button_disabled = () => <Button disabled>Mon Bouton</Button>;
export const Button_with_style_props = () => <Button style={{color: 'blue', backgroundColor: 'pink', fontSize: '2rem', padding: '2rem'}}>Mon Bouton</Button>;
export const Button_with_node_children = () => <Button><h1>Mon Bouton en h1</h1></Button>;
