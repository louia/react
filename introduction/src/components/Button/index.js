import React from 'react';
import PropTypes from 'prop-types';

export function Button(props) {
    return (
        <button {...props}><h1>{props.children}</h1></button>
    );
}

export default Button;
