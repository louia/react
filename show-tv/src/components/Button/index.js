import React from 'react';
import PropTypes from 'prop-types';

export class Button extends React.Component {
    render() {
        if (this.props.actif == true) {
            return (
                <button className="actif">{this.props.children}</button>
            );
        }
        return (
            <button className="inactif" disabled>{this.props.children}</button>
        );
    }
}

Button.propTypes = {
    actif: PropTypes.bool,
};


export default Button;
