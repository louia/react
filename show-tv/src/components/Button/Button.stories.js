import React from 'react';
import Button from './index';
import '../../index.css'

export default {
    title: 'components/Button',
    component: Button,
}

export const default_value = () => <Button />;
export const initialized = () => <Button actif={false}  children="Button désactivé"/>;
export const actif = () => <Button actif={true}  children="Button activé"/>;
