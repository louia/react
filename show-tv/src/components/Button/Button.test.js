import { render, fireEvent } from '@testing-library/react';
 import { default_value, initialized } from './Button.stories';
 


describe('Button test', () => {
    test('should increment when clicked', () => {
        expect(true).toBeTruthy();
    });

});
