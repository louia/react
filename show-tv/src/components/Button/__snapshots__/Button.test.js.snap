// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components/Button Actif 1`] = `
<button
  className="inactif"
/>
`;

exports[`Storyshots components/Button Default Value 1`] = `
<button
  className="inactif"
/>
`;

exports[`Storyshots components/Button Inactif 1`] = `
<button
  className="actif"
>
  Button
</button>
`;

exports[`Storyshots components/Button Initialized 1`] = `
<button
  className="actif"
>
  Button
</button>
`;

exports[`Storyshots components/Button Kjfd 1`] = `
<button
  className="actif"
>
  Button
</button>
`;

exports[`Storyshots components/Header Default Value 1`] = `
<header>
  <div
    className="item-header"
  />
  <div
    className="item-header"
  />
</header>
`;

exports[`Storyshots components/Header Initialized 1`] = `
<header>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;

exports[`Storyshots components/Header Mini 1`] = `
<header
  className="mini"
>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;
